### DISCUSSION TITLE
How to run clang-tidy with same settings as clangs with query-driver driver argument?

### DISCUSSION THREAD

**Discussion by KindDragon** [clangd/clangd#1368] [Q&A]
I run clangd using VSCode plugin and use there `--query-driver=${env:TOOLCHAIN_PATH}/bin/arm-none-eabi-g++` argument, and also I have .clang config with following configuration:
```
CompileFlags:
  Remove:
    - -fno-aggressive-loop-optimizations
```
Clang-Tidy inside VSCode works fine with this configuration.

How can I run clang-tidy from the command line with the same configuration?

**Comment by i-ky** [comment]
Unfortunately, clang-tidy has no equivalents of `--query-driver` and `Remove:`, see [docs](https://clang.llvm.org/extra/clang-tidy/#using-clang-tidy). Perhaps, you can mimic `--query-driver` by manually specifying necessary include directories via `--extra-arg` or `--extra-arg-before`, but given that you need to remove an option clang does not support, you will likely need to massage compilation database with scripts. If you plan heavy usage of clang-based tools with your project, I would recommend "teaching" your build system to compile your project with clang, at least for the sake of producing clang-compatible compilation database.

**Reply by KindDragon** [reply]
Thanks, also I found this workaround for`--query-driver` https://github.com/llvm/llvm-project/issues/53468#issuecomment-1025128992
